src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extensions.cs

```diff
@@ -0,0 +1,539 @@
+                var underlyingTypeWithAnnotations = underlyingTypeBinder.BindType(underlyingTypeSyntax, diagnostics, basesBeingResolved);
+
+                TypeSymbol underlyingType = underlyingTypeWithAnnotations.Type;
+                checkStatic(diagnostics, location, underlyingType);
+                if (IsRestrictedExtensionUnderlyingType(underlyingTypeWithAnnotations))
+                {
+                    diagnostics.Add(ErrorCode.ERR_BadExtensionUnderlyingType, location, this, underlyingTypeWithAnnotations);
+                }
+                else
+                {
+                    partUnderlyingType = underlyingType;
+                }
+            }
+            else
+            {
+                sawUnderlyingType = false;
+            }
+
+            var bases = syntax.BaseList;
+            var partBaseExtensions = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+            if (bases != null)
+            {
+                Binder baseBinder = this.DeclaringCompilation.GetBinder(bases);
+                baseBinder = adjustBinder(syntax, baseBinder);
+
+                foreach (var baseTypeSyntax in bases.Types)
+                {
+                    TypeSyntax typeSyntax = baseTypeSyntax.Type;
+                    var location = new SourceLocation(typeSyntax);
+
+                    TypeWithAnnotations baseTypeWithAnnotations = baseBinder.BindType(typeSyntax, diagnostics, basesBeingResolved);
+                    TypeSymbol baseType = baseTypeWithAnnotations.Type;
+                    checkStatic(diagnostics, location, baseType);
+
+                    switch (baseType.TypeKind)
+                    {
+                        case TypeKind.Extension when baseTypeWithAnnotations.NullableAnnotation != NullableAnnotation.Annotated:
+                            foreach (var baseExtension in partBaseExtensions)
+                            {
+                                ReportDuplicateLocally(baseExtension, baseType, location, diagnostics, forBaseExtension: true);
+                            }
+                            partBaseExtensions.Add((NamedTypeSymbol)baseType);
+                            break;
+
+                        case TypeKind.Error:
+                            partBaseExtensions.Add((NamedTypeSymbol)baseType);
+                            break;
+
+                        default:
+                            diagnostics.Add(ErrorCode.ERR_OnlyBaseExtensionAllowed, location, baseTypeWithAnnotations);
+                            break;
+                    }
+                }
+            }
+
+            return new ExtensionInfo(partUnderlyingType, partBaseExtensions.ToImmutableAndFree(),
+                isExplicit: syntax.ImplicitOrExplicitKeyword.IsKind(SyntaxKind.ExplicitKeyword));
+
+            void checkStatic(BindingDiagnosticBag diagnostics, Location location, TypeSymbol baseType)
+            {
+                if (baseType.IsStatic && !this.IsStatic)
```

Comment:
>if (baseType.IsStatic && !this.IsStatic) [](http://example.com/codeflow?start=16&length=40)

Is ```!baseType.IsStatic && this.IsStatic``` a valid combination?

Reply:
A static extension on an instance underlying type and any kind of base extension (either instance or static) seems fine.
Such an extension will be restricted to having static members and prevented from using `this`. There will also be some impact on consumers (usage like `StaticExtension e = underlyingValue;` would likely be disallowed).

I can add an open question to spec if you'd like.